packages/cactus-plugin-keychain-azure-kv/src/main/typescript/credentials/refresh-token-credential.ts

```diff
@@ -0,0 +1,117 @@
+  readonly refreshToken: Readonly<string>;
+  readonly resource: Readonly<string>;
+}
+
+export class RefreshTokenCredential implements TokenCredential {
+  public static readonly CLASS_NAME = "RefreshTokenCredential";
+
+  private readonly log: Logger;
+
+  public get className(): string {
+    return RefreshTokenCredential.CLASS_NAME;
+  }
+  private readonly msalClient: Readonly<PublicClientApplication>;
+
+  constructor(private readonly opts: Readonly<IRefreshTokenCredentialOptions>) {
+    const fn = `${this.className}#constructor()`;
+    if (!opts) {
+      throw new Error(`${fn} - arg opts cannot be falsy.`);
+    }
+    const { authority, clientId, refreshToken, resource } = opts;
+    const { logLevel = "DEBUG" } = opts;
+
+    // verify that the required arguments were passed in
+    if (!authority) {
+      throw new Error(`${fn} - arg opts.authority cannot be falsy.`);
+    }
+    if (!clientId) {
+      throw new Error(`${fn} - arg opts.clientId cannot be falsy.`);
+    }
+    if (!refreshToken) {
+      throw new Error(`${fn} - arg opts.refreshToken cannot be falsy.`);
+    }
+    if (!resource) {
+      throw new Error(`${fn} - arg opts.resource cannot be falsy.`);
+    }
+
+    // verify that the arguments are of string type
+    if (typeof authority !== "string") {
+      throw new Error(`${fn} - arg opts.authority must be string.`);
+    }
+    if (typeof clientId !== "string") {
+      throw new Error(`${fn} - arg opts.clientId must be string.`);
+    }
+    if (typeof refreshToken !== "string") {
+      throw new Error(`${fn} - arg opts.refreshToken must be string.`);
+    }
+    if (typeof resource !== "string") {
+      throw new Error(`${fn} - arg opts.resource must be string.`);
+    }
+
+    // Verify if the strings are empty or only contain whitespaces
+    if (clientId.trim().length < 1) {
+      throw new Error(`${fn} - arg opts.clientId cannot be blank.`);
+    }
+    if (refreshToken.trim().length < 1) {
+      throw new Error(`${fn} - arg opts.refreshToken cannot be blank.`);
+    }
+    if (authority.trim().length < 1) {
+      throw new Error(`${fn} - arg opts.authority cannot be blank.`);
+    }
+    if (resource.trim().length < 1) {
+      throw new Error(`${fn} - arg opts.resource cannot be blank.`);
+    }
+
+    this.log = LoggerProvider.getOrCreate({
+      label: this.className,
+      level: logLevel,
+    });
+    this.log.debug("Creating PublicClientApplication...");
+
+    this.msalClient = new PublicClientApplication({
+      auth: {
+        clientId,
+        authority,
+        clientSecret: refreshToken,
+      },
+    });
+    this.log.debug("Creating PublicClientApplication OK");
+    this.log.debug("EXIT");
+  }
+
+  async getToken(): Promise<AccessToken | null> {
+    const fn = `${this.className}#getToken()`;
+    const result = await this.msalClient.acquireTokenByRefreshToken({
+      refreshToken: this.opts.refreshToken,
+      scopes: [`${this.opts.resource}/.default`],
```

Comment:
Shall we later improvise upon passing in of scopes as an argument or read them from env vars?

Reply:
@jagpreetsinghsasan I've added the parameter for the scopes!